src/Controls/tests/DeviceTests/Elements/TextInput/TextInputTests.cs

```diff
@@ -1,24 +1,338 @@
+		{
+			string text = "This is TEXT!";
+
+			var editor = new TView
+			{
+				Text = text,
+				CursorPosition = initialPosition
+			};
+
+			await ValidatePropertyInitValue<int, THandler>(
+				editor,
+				() => editor.CursorPosition,
+				GetPlatformCursorPosition,
+				text.Length);
+		}
+
+		[Theory(DisplayName = "Unset CursorPosition is kept at zero at initialization")]
+		[InlineData("This is a test!!!")]
+		[InlineData("a")]
+		[InlineData("")]
+		[InlineData(" ")]
+		public async Task UnsetCursorPositionKeepsToZeroOnInitialization(string text)
+		{
+			var editor = new TView
+			{
+				Text = text
+			};
+
+			await ValidatePropertyInitValue<int, THandler>(
+				editor,
+				() => editor.CursorPosition,
+				GetPlatformCursorPosition,
+				0);
+		}
+
+		[Theory(DisplayName = "CursorPosition moves to the end on text change after initialization")]
+		[InlineData("This is a test!!!")]
+		[InlineData("a")]
+		[InlineData("")]
+		[InlineData(" ")]
+		public async Task CursorPositionMovesToTheEndOnTextChangeAfterInitialization(string text)
+		{
+			var editor = new TView
+			{
+				Text = "Test"
+			};
+
+			await AttachAndRunFocusAffectedControl<TView, THandler>(editor, handler => editor.Text = text);
+
+			Assert.Equal(text.Length, editor.CursorPosition);
+		}
+
+		[Theory(DisplayName = "SelectionLength Updates Correctly")]
+		[InlineData(2, 5)]
+		public async Task SelectionLengthUpdatesCorrectly(int setValue, int unsetValue)
+		{
+			string text = "This is TEXT!";
+
+			var editor = new TView
+			{
+				Text = text,
+			};
+
+			await ValidatePropertyUpdatesValue<int, THandler>(
+				editor,
+				nameof(ITextInput.SelectionLength),
+				GetPlatformSelectionLength,
+				setValue,
+				unsetValue
+			);
+		}
+
+		[Theory(DisplayName = "SelectionLength is Capped to Text Length")]
+		[InlineData(30)]
+		public async Task SelectionLengthIsCapped(int selectionLength)
+		{
+			string text = "This is TEXT!";
+
+			var editor = new TView
+			{
+				Text = text,
+				SelectionLength = selectionLength
+			};
+
+			await ValidatePropertyInitValue<int, THandler>(
+				editor,
+				() => editor.SelectionLength,
+				GetPlatformSelectionLength,
+				text.Length);
+		}
+
+		[Theory(DisplayName = "Unset SelectionLength is kept at zero at initialization")]
+		[InlineData("This is a test!!!")]
+		[InlineData("a")]
+		[InlineData("")]
+		[InlineData(" ")]
+		public async Task UnsetSelectionLengthKeepsToZeroOnInitialization(string text)
+		{
+			var editor = new TView
+			{
+				Text = text
+			};
+
+			await ValidatePropertyInitValue<int, THandler>(
+				editor,
+				() => editor.SelectionLength,
+				GetPlatformSelectionLength,
+				0);
+		}
+
+		[Theory(DisplayName = "SelectionLength is kept at zero on text change after initialization")]
+		[InlineData("This is a test!!!")]
+		[InlineData("a")]
+		[InlineData("")]
+		[InlineData(" ")]
+		public async Task SelectionLengthMovesToTheEndOnTextChangeAfterInitialization(string text)
```

Comment:
```suggestion
		public async Task SelectionLengthKeepsToZeroOnTextChangeAfterInitialization(string text)
```

Reply:
I don't think this test simulates the action of a user changing the text by updating the entry text manually themselves. I think it just simulates the action of the developer setting new text programatically. And that is why the cursor position remains at 0 - because the text doesn't exactly "change" in that sense